core/src/consensus.rs

```diff
@@ -156,20 +159,35 @@ impl TowerVersions {
                     node_pubkey: tower.node_pubkey,
                     threshold_depth: tower.threshold_depth,
                     threshold_size: tower.threshold_size,
-                    vote_state: tower.vote_state,
+                    vote_state: VoteStateVersions::V1_10_40(Box::new(tower.vote_state))
+                        .convert_to_current(),
                     last_vote: box_last_vote,
                     last_vote_tx_blockhash: tower.last_vote_tx_blockhash,
                     last_timestamp: tower.last_timestamp,
                     stray_restored_slot: tower.stray_restored_slot,
                     last_switch_threshold_check: tower.last_switch_threshold_check,
                 }
             }
+
+            TowerVersions::V1_10_40(tower) => Tower {
+                node_pubkey: tower.node_pubkey,
+                threshold_depth: tower.threshold_depth,
+                threshold_size: tower.threshold_size,
+                vote_state: VoteStateVersions::V1_10_40(Box::new(tower.vote_state))
+                    .convert_to_current(),
```

Comment:
Yes convert_to_current is the point where an older vote state is detected and upgraded to the new vote state.  So this must be called when the tower is loaded in case the tower's vote state was the older form.